### ISSUE TITLE
Bundled logger doesn't print error when NODE_ENV is set

### ISSUE THREAD

**Issue by turacma** [medusajs/medusa#9862]
### Describe the bug

When `NODE_ENV` is set to production/staging, the default logger switches to using JSON formatting, but when passed a error as part of a catch statement doesn't actually print the error details.

### System information

Medusa version (including plugins):
Node.js version: 20.15.1
Database: Postgres
Operating system: Mac OS
Browser (if relevant): N/A

### Steps to reproduce the behavior

1. Create a new medusa installation
2. Configure as normal
3. Break the configuration such that it won't start, note that the actual error is logged
4. run `export NODE_ENV=staging` or `export NODE_ENV=production` then start again, note that error is not printed, only stack trace and the message.

### Expected behavior

Ideally I would be able to provide my own logger implementation to use, but failing that, the bundled logger should at least print out the error details when providered.

### Screenshots

N/A

### Code snippets

N/A

### Additional context

Not sure if related to my configuration or a general problem, but also noticing that when `NODE_ENV=staging|production` server fails to start up with the error

```
Error starting server Error: Could not find index.html in the admin build directory. Make sure to run 'medusa build' before starting the server.
    at serve (/Users/turacma/Workspace/ship/node_modules/@medusajs/admin-bundler/dist/index.js:1344:11)
    at serveProductionBuild (/Users/turacma/Workspace/ship/node_modules/@medusajs/medusa/src/loaders/admin.ts:75:28)
    at async loadEntrypoints (/Users/turacma/Workspace/ship/node_modules/@medusajs/medusa/src/loaders/index.ts:108:3)
    at async exports.default (/Users/turacma/Workspace/ship/node_modules/@medusajs/medusa/src/loaders/index.ts:164:31)
    at async internalStart (/Users/turacma/Workspace/ship/node_modules/@medusajs/medusa/src/commands/start.ts:120:50)
    at async start (/Users/turacma/Workspace/ship/node_modules/@medusajs/medusa/src/commands/start.ts:221:5)
```

**Comment by turacma** [comment]
Forking the additional context to a new issue report

**Comment by thetutlage** [comment]
Hello @turacma 

I just gave it a try by pointing a non-existing module and everything looks fine to me as per the shared screenshot. Are you seeing things differently?

![CleanShot 2024-11-04 at 12 20 23@2x](https://github.com/user-attachments/assets/18c89397-0fe6-4ad1-a678-4b99093fe843)

**Comment by thetutlage** [comment]
Also, its worth mentioning. If you are running the production build (ie, the JavaScript output). You must `cd` into the `.medusa/server` directory and run it from there.

**Comment by turacma** [comment]
> Also, its worth mentioning. If you are running the production build (ie, the JavaScript output). You must cd into the .medusa/server directory and run it from there.

Yep, this was noted in my other bug report. As noted there, FWIW, this didn't seem to be necessary with the preview build I was using previously.

Here's an example of breaking something in the middleware:

![-zsh](https://github.com/user-attachments/assets/55c2684c-1ceb-424f-b871-134f7e67b114)
![-zsh](https://github.com/user-attachments/assets/bb6b8093-327f-455c-95c0-5d0418ce19af)

Might be be because it's a nested exception?

**Comment by thetutlage** [comment]
> Might be be because it's a nested exception

Okay, yes. The `cause` property is not entertained when the JSON output is used. Sure, this is something we will look into and fix. 

Thanks for reporting the issue :)

**Comment by thetutlage** [comment]
Fixed by: https://github.com/medusajs/medusa/pull/9944
